views.py

```diff
@@ -1081,6 +1086,210 @@ class RegistrationActiveView(HTTPMethodView):
+
+        return json({'msg': 'Feedback association created successfully'}, 200)
+
+    @user_required('admin')
+    async def delete(self, _, current_user, qid, lid):
+        try:
+            # doing this twice since delete() could throw a weird out of bounds exc.
+            await LessonFeedbackMeta.get_first_by_many_field_value(
+                question=qid,
+                lesson=lid
+            )
+            await LessonFeedbackMeta.delete_by_many_fields(
+                question=qid,
+                lesson=lid
+            )
+
+            return json({}, 204)
+        except DoesNotExist:
+            return json({'error': 'No association with given constraints exists in the database'}, 404)
+
+
+class LessonFeedbackAnswerView(HTTPMethodView):
+    async def _validate_answers(self, data):
+        provided_answers = data['answers']
+        question = None
+
+        try:
+            question = await LessonFeedbackQuestion.get_first("id", data['question'])
+        except DoesNotExist:
+            return json({'error': 'There is no question with given id'}, 400)
+
+        if question.type == 'abcd_single':
+            available_answers = jloads(question.answers)
+
+            if jloads(provided_answers)[0] not in available_answers:
+                return INVALID_ANSWER
+        elif question.type == 'abcd_multiple':
+            available_answers = jloads(question.answers)
+
+            if set(jloads(provided_answers)) - set(available_answers):
+                return INVALID_ANSWER
+        elif question.type == 'int':
+            try:
+                int(provided_answers)
+            except ValueError:
+                return INVALID_ANSWER
+
+    @user_required()
+    async def get(self, _, current_user, lid):
+        try:
+            await Lesson.get_first("id", lid)
+        except DoesNotExist:
+            return json({'error': 'There is no lesson with given id'}, 404)
+
+        author = current_user.id
+        answers = await LessonFeedbackAnswer.get_by_many_field_value(
+            author=author,
+            lesson=lid
+        )
+
+        return json(answers)
+
+    @user_required()
+    async def post(self, request, current_user):
+        required_fields = {'answers', 'question', 'lesson'}
+        data = request.json
+        if set(data.keys()) != required_fields:
+            return json({'error': 'Missing required field(s)'}, 400)
+
+        response = await self._validate_answers(data)
+
+        if response:
+            return response
+
+        data.update({'author': current_user.id})
+
+        answer = LessonFeedbackAnswer(**data)
+        await answer.create()
+
+        return json({'msg': 'Answer recorded'})
+
+    @user_required()
+    async def put(self, request, current_user, aid):
+        available_fields = {'answers', 'question', 'lesson'}
+        data = request.json
+        answer = None
+
+        try:
+            answer = await LessonFeedbackAnswer.get_first("id", aid)
```

Comment:
there is get by id method :)

Reply:
There is, but it doesn't do what I want it to IIRC - it throws an out of bounds-like exception instead of a DoesNotExist one.